constraint/pkg/client/drivers/k8scel/transform/cel_snippets.go

```diff
@@ -169,6 +169,23 @@ func BindParamsCEL() []cel.NamedExpressionAccessor {
 	}
 }
 
+func BindObjectV1Beta1() admissionregistrationv1beta1.Variable {
+	return admissionregistrationv1beta1.Variable{
+		Name:       schema.ObjectName,
+		Expression: `has(request.operation) && request.operation == "DELETE" && object == null ? oldObject : object`,
```

Comment:
This logic for rego is actually in GK not framework: https://github.com/open-policy-agent/gatekeeper/blob/f81f0acacb00802108abc81ba5d862bec6c2d633/pkg/util/request_validation.go#L29, should it be moved to framework to be consistent? not a PR blocker.

Reply:
Framework should not have any coupling with any particular platform.

We should probably move the logic to the TargetHandler.

In addition, we should update the engine logic to __undo__ the setting of obj == oldObject on DELETE, otherwise CEL code will behave differently in on-k8s VAP vs in-g8r VAP.

Reply:
> We should probably move the logic to the TargetHandler.

Move the `obj == oldObject on DELETE` logic to the TargetHandler instead of the webhook validation handler right?

Reply:
WRT moving it to target handler -- right.

The problem with moving it to the Rego driver is that the Rego driver is more generic than the VAP driver.

The VAP driver is highly coupled with K8s/KRM -- it is strongly typed, assuming an inbound admission review object. Because of this, and because the VAP driver is explicitly intended to support VAP (as opposed to a more generic application of CEL) I'm okay with adding "undoing" logic to the driver (we're not increasing coupling).

For something like the Rego driver, the only real assumption is that the inbound object is JSON-structured.